### PR TITLE
Feature/share storage

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: contracts/OracleStorage.sol [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    // Event
+    event ValueUpdated(
+        bytes32 indexed id,
+        bytes32 indexed oracleCallID,
+        uint256 date,
+        bytes value
+    );
+
+    // Update value
+    function updateValue(bytes32 id, bytes32 callId, uint256 date, bytes memory value)
```
Comment (Amxx) [inline contracts/OracleStorage.sol:40]: IMO, when a function is internal it should be prefixed with an underscore

```suggestion
    function _updateValue(bytes32 id, bytes32 callId, uint256 date, bytes memory value)
```

That way it is clear from the calling contract that this is an "unprotected" internal operation

File: contracts/GenericOracle.sol [hunk 2]
```diff
@@ -68,45 +54,7 @@ contract GenericOracle is IexecDoracle, Ownable, IOracleConsumer {
...
-    }
-
-    function getString(bytes32 _oracleId)
-        public
-        view
-        returns (string memory stringValue, uint256 date)
-    {
-        bytes memory value = values[_oracleId].value;
-        return (abi.decode(value, (string)), values[_oracleId].date);
+        require(updateValue(id, _callID, date, value), "Failed to update value");
```
Comment (Amxx) [inline contracts/GenericOracle.sol:57]:
```suggestion
        require(_updateValue(id, _callID, date, value), "Failed to update value");
```

File: contracts/OracleStorage.sol [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        internal
+        returns (bool)
+    {
+        values[id].date = date; //What if date is older?
+        values[id].value = value;
+        emit ValueUpdated(id, callId, date, value);
+        return true;
```
Comment (Amxx) [inline contracts/OracleStorage.sol:47]:
```suggestion
    {
        values[id].date = date; //What if date is older?
        values[id].value = value;
        emit ValueUpdated(id, callId, date, value);
```

```suggestion
```

This function cannot revert. No need to return a bool

File: contracts/OracleStorage.sol (hunk 1 above)
Comment (jeremyjams) [reply contracts/OracleStorage.sol]: Thank you :pray:

File: contracts/OracleStorage.sol (hunk 3 above)
Comment (jeremyjams) [reply contracts/OracleStorage.sol]: I was planning to add a require at some point for `//What if date is older?`, but I agree, there is no need right now.
Thanks, removed and updated (https://github.com/iExecBlockchainComputing/generic-oracle-contracts/pull/7/commits/5456953c9f203d5caef17ee95fca59a25fcf0835) :)

File: contracts/GenericOracle.sol [hunk 4]
```diff
@@ -16,14 +16,10 @@
...
  ******************************************************************************/
 
-pragma solidity ^0.6.12;
-pragma experimental ABIEncoderV2;
+pragma solidity >=0.6.12;
 
-import "@openzeppelin/contracts/access/Ownable.sol";
-import "@iexec/solidity/contracts/ERC1154/IERC1154.sol";
-import "@iexec/doracle/contracts/IexecDoracle.sol";
+contract GenericOracle {
```
Comment (zguesmi) [inline contracts/GenericOracle.sol:21]: How about?
```suggestion
abstract contract GenericOracle {
```

**Review by zguesmi — APPROVED** [review]
I find this version clearer. My comment is optional if time is a constraint.

File: contracts/GenericOracle.sol (hunk 4 above)
Comment (jeremyjams) [reply contracts/GenericOracle.sol]: Good idea :+1: Please see these 3 commits https://github.com/iExecBlockchainComputing/generic-oracle-contracts/pull/7/files/09e92659f35fe9390d09a4b24d741b8e93642f1f..ccccf525c3553cbe77f0c61a74e9a5b795b9f867

**Pull request merged**